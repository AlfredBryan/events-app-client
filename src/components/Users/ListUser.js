import React, { Component } from "react";
import axios from "axios";

import NavBar from "../NavBar/NavBar";
import "./User.css";

const apiURL = "http://localhost:4000/api/users/all";

class ListUser extends Component {
  state = {
    users: []
  };

  componentDidMount() {
    axios
      .get(`${apiURL}`)
      .then(res => {
        this.setState({
          users: res.data
        });
      })
      .catch(error => {
        throw error;
      });
  }

  render() {
    let { users } = this.state;
    return (
      <React.Fragment>
        <NavBar />
        <div className="user-main">
          <h5>List of Registered Users</h5>
          {users.map(user => (
            <div key={user.id} className="d-flex wrap">
              <p className="fname">{user.firstName}</p>
              <p className="lname">{user.lastName}</p>
            </div>
          ))}
        </div>
      </React.Fragment>
    );
  }
}

export default ListUser;
