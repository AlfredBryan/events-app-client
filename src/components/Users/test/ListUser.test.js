import React from "react";
import ListUser from "../ListUser";
import { shallow } from "enzyme";

jest.mock("axios", () => {
  const testUsers = {
    users: []
  };

  return {
    get: jest.fn(() => Promise.resolve(testUsers))
  };
});

const axios = require("axios");

it("Fetch registered Users #componentDidMount", done => {
  const wrapper = shallow(<ListUser />);
  wrapper.instance().componentDidMount();
  expect(axios.get).toHaveBeenCalled();
  expect(axios.get).toHaveBeenCalledWith(
    "https://events-apps.herokuapp.com/api/users/all"
  );
  expect(wrapper.state()).toMatchSnapshot("users", []);
  done();
});
