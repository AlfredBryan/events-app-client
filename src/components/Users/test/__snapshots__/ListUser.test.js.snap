// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch registered Users #componentDidMount: users 1`] = `
Object {
  "loading": true,
  "users": Array [],
}
`;
